packages/view-types/link-controller/src/LinkController.js

```diff
@@ -0,0 +1,143 @@
+import React, {useMemo, useEffect, useRef, useCallback, useState} from 'react';
+import {
+  TitleInfo,
+  useViewConfigStoreApi,VitS
+} from '@vitessce/vit-s';
+import {
+  FormControlLabel, Switch, Box, Checkbox, Grid
+} from '@material-ui/core';
+import {createLoaders} from "@vitessce/vit-s/src/vitessce-grid-utils.js";
+
+
+export default function LinkController(props) {
+  const {
+    linkIDInit,viewTypes, fileTypes, coordinationTypes, stores
+  } = props
+  const viewConfigStoreApi = useViewConfigStoreApi();
+
+  console.log(viewTypes, fileTypes, coordinationTypes, stores)
+
+  const [socketOpen, setSocketOpen] = useState(false);
+  const [sync, setSync] = useState(true)
+  const connection = useRef(null);
+  const [linkID, setLinkID] = useState(null)
+  const id = useMemo(() => {
+    return 'id_' + (Math.floor(Math.random() * 1000));
+  }, []);
+
+  // Reference: https://github.com/pmndrs/zustand/tree/v3.7.2?tab=readme-ov-file#readingwriting-state-and-reacting-to-changes-outside-of-components
+  //const config = viewConfigStoreApi.getState().viewConfig;
+  const setConfig = useCallback((newConfig) => {
+    console.log(fileTypes,coordinationTypes,stores)
+    viewConfigStoreApi.setState({viewConfig: newConfig, mostRecentConfigSource: 'external', loaders: createLoaders(
+        newConfig.datasets,
+        newConfig.description, fileTypes, coordinationTypes, stores
+      )});
+  }, [viewConfigStoreApi]);
+
+  const onConfigChange = useCallback((configValue) => {
+    // console.log(send, socketOpen)
+    if (sync && socketOpen) {
+      try {
+        //console.log(configValue)
+        let configString = id + ';' + btoa(escape(JSON.stringify(configValue)));
```

Comment:
removing escape or using `encodeURIComponent` to keep consistent across Vitessce?